src/components/TableCard/TableCard.jsx

```diff
@@ -208,6 +209,50 @@ const determinePrecisionAndValue = (precision = 0, value, locale) => {
   return '--';
 };
 
+/**
+ * Updates the hrefs in each column to be us-able links. If href variables are on a table that has row specific values, the user
+ * should not pass in a cardVariables object as each variable with have multiple values.
+ * @param {array} columns - Array of TableCard columns
+ * @param {object} cardVariables - object of cardVariables
+ * @return {array} array of columns with formatted links and updated variable values
+ */
+export const createColumnsWithFormattedLinks = (columns, cardVariables) => {
+  return columns.map(column => {
+    const { linkTemplate } = column;
+    if (linkTemplate) {
+      let variables;
+      if (!cardVariables) {
+        // fetch variables on the href
+        variables = linkTemplate.href ? getVariables(linkTemplate.href) : [];
+      }
+      return {
+        ...column,
+        // eslint-disable-next-line react/prop-types
+        renderDataFunction: ({ value, row }) => {
+          let variableLink;
+          // if we have variables the value based on its own row's context
+          if (variables && variables.length) {
```

Comment:
How did you validate that this works? `variables` isn't defined as a param, and I think is outside the scope of the function, no?

Reply:
@tay1orjones That is what I thought at first too, but it seems to work fine. Verified with Monitor's use-case when linking my local addons repo to monitor. Also verified in the new TableCard story I have added. I was not able to break it based on variable scope.

Reply:
Must be the syntax tripping me up. The arrow function binds the entire lexical context, which includes all the variables in the current scope. Glad you verified via `link` too!